### PR TITLE
refactor language index and subindex grid layout

### PR DESCRIPTION
**Pull request by jdamiba**
The purpose of this PR is to refactor `documentation_eg.html` in order to only display the first 5 examples on the language index page while displaying all examples on subindex pages. 

As a result of this refactor, the `display_as:example_index` front-matter is no longer used to determine whether a post is displayed on the language index page.

### CODE REVIEW

**Comment by nicolaskruchten** [comment]
The approach is good! Can you take a look at the Percy diffs and fix up some of the bigger undesirable changes in Matlab/ggplot?

maybe restrict this order check just to the Big 3 languages?

**Comment by jdamiba** [comment]
Ok, working on making sure only the R, Python, and JS docs are affected by this.

**Comment by jdamiba** [comment]
@nicolaskruchten I have updated this PR per our conversation this afternoon. Restricting the order changes to Python, R, and JS was accomplished by creating a `mainlang_documentation_eg.html` include which is only used for those languages' index pages.

File: _posts/ggplot2/2016-12-16-ggplot2-index.md [hunk 1]
```diff
@@ -20,6 +20,6 @@ redirect_from: ggplot2/reference/
   </div>
 </header>
 
-{% assign languagelist = site.posts | where:"page_type","example_index" | where:"language","ggplot2"  | sort: "order"  %}
+{% assign languagelist = site.posts | where:"page_type","example_index" | where:"language","ggplot2" | sort: "order" %}
```
Comment (nicolaskruchten) [inline _posts/ggplot2/2016-12-16-ggplot2-index.md:23]: so for ggplot2, given that we don't have sub-index pages, we should remove the `example_index` filter also, right?

File: _posts/julia/2015-04-05-julia-index.html [hunk 2]
```diff
@@ -4,6 +4,7 @@
 description: Plotly's Julia graphing library makes interactive, publication-quality graphs online. Examples of how to make line plots, scatter plots, area charts, bar charts, error bars, box plots, histograms, heatmaps, subplots, multiple-axes, polar charts and bubble charts.
 layout: langindex
 language: julia
+display_as: false
```
Comment (nicolaskruchten) [inline _posts/julia/2015-04-05-julia-index.html:7]: we only need this line for the big three now, right?

**Comment by nicolaskruchten** [comment]
Good cleanup!

So there are a few ordering issues that can be resolved in this repo, if you look at the JS index page and the R/Python Chart Studio pages.

Also, we should manually clear up the order for the random sections in Python and R like chart-events etc, so that the top 5 at least have the correct orders. Let's merge those changes first so that this diff looks good before we merge it.

**Comment by jdamiba** [comment]
superseded by https://github.com/plotly/graphing-library-docs/pull/78

**Pull request closed without merging**